### DISCUSSION TITLE
How to get file value before reset?

### DISCUSSION THREAD

**Discussion by Hansum** [react-hook-form/react-hook-form#2187] [Q&A]
Hi, sorry for this stupid question but how can i get the `<input type="file" ref={register} name="image"/>` value before resetting the form inputs? I've read the documentation but maybe I just miss something. Thank you.

[CSB](https://codesandbox.io/s/react-hook-form-reset-v6-d21qc?file=/src/index.js)

**Comment by bluebill1049** [comment]
i haven't looked at your codesandbox, but you can use `getValues()` before the reset()

**Reply by Hansum** [reply]
I tried to delete `e.target.reset` and it works like a charm and got the input value. Is there anyway to reset the all the input after submitting the form? Thank you for the fast response btw.

**Reply by bluebill1049** [reply]
probably need to sort the input values in a variable then.

**Reply by Hansum** [reply]
Thank you for the response @bluebill1049 but I have another question. Why does the file input returns undefined when I try to use `reset()` but reset it ignores other input values?. Sorry for asking stupid questions, I'm just lost right now.

**Reply by bluebill1049** [reply]
that's all good mate :) file object is immutable for security reason.
